api_tests/lib_sdk/wallets/lightning.ts

```diff
@@ -82,4 +82,73 @@ export class LightningWallet implements Wallet {
     public addPeer(toWallet: LightningWallet) {
         return this.inner.addPeer(toWallet.inner);
     }
+
+    public getPeers() {
+        return this.inner.getPeers();
+    }
+
+    public getChannels() {
+        return this.inner.getChannels();
+    }
+
+    public async openChannel(toWallet: LightningWallet, quantity: number) {
+        // First, need to check everyone is sync'd to the chain
+
+        const thisIsSynced = (await this.inner.getWalletInfo())
+            .is_synced_to_chain;
+        const toIsSynced = (await toWallet.inner.getWalletInfo())
+            .is_synced_to_chain;
+
+        while (!thisIsSynced || !toIsSynced) {
+            this.logger.info(
+                `One of the lnd node is not yet synced, waiting. this: ${thisIsSynced}, to: ${toIsSynced}`
+            );
+            await sleep(500);
+        }
```

Comment:
📣
I think you have to to recheck if it is synched in the loop? Otherwise this will just keep looping.